src/lib.rs

```diff
@@ -89,16 +82,15 @@ impl EncodingDetector {
 
     /// Pass a chunk of raw bytes to the detector. This is a no-op if a
     /// charset has been detected.
-    fn handle_data(&mut self, data: &[u8]) -> EncodingDetectorResult<()> {
+    fn handle_data(&mut self, data: &[u8]) -> Result<()> {
         let result = unsafe {
             ffi::uchardet_handle_data(self.ptr, data.as_ptr() as *const i8,
                                       data.len() as size_t)
         };
         match result {
             0 => Ok(()),
             _ => {
-                let msg =  "Error handling data".to_string();
-                Err(EncodingDetectorError{message: msg})
+                Err(ErrorKind::DataHandlingError.into())
```

Comment:
Are there any other uchardet error codes we should map to separate `ErrorKind` values? Or does it just return one cryptic error code for everything?